src/firefly/js/tables/ui/PropertySheet.jsx

```diff
@@ -0,0 +1,50 @@
+import {getActiveTableId, getTblById, tableDetailsView} from '../TableUtil.js';
+import {TablePanel} from 'firefly/tables/ui/TablePanel';
+import React, {useEffect} from 'react';
+import {useStoreConnector} from 'firefly/ui/SimpleComponent';
+import {dispatchTableAddLocal} from 'firefly/tables/TablesCntlr';
+
+export function PropertySheet({tbl_group='main', tbl_id, children, fetcher}){
+    const {activeTblId, tableModel} = useStoreConnector(() => {
+        const tblId = tbl_id || getActiveTableId(tbl_group);
+        return {activeTblId: tblId, tableModel: getTblById(tblId)};
+    });
```

Comment:
We talked about using children as a `renderer` and that it should default to <PropertySheetTable {...{tbl_id, highlightedRow}}/>.

In the most basic case: (RowDetails of the active table)
```jsx
  <PropertySheet/>
```

In FIREFLY-603, it mentioned templating. the idea is this.
```jsx
  <PropertySheet>
    <CustomView/>
  </PropertySheet>

function CustomView(p) {
return <div> Hello, ${p.firstName} ${p.lastName}</div>;
}
```
In this scenario, p is the return value of the default `fetcher` function where it transform the highlightedRow into an object of cname/value pairs.  This object is passed in as properties to the renderer.

Then, there's the more advanced use-case where fetcher is used to query external data.
In this case, fetcher is most likely async.  `PropertySheet` should call `fetcher` with `tbl_id`, and `highlightedRow` so it can gather additional table meta.  It should also be smart enough to render 'Loading.. ' status until `fetcher` returns.

For this PR(FIREFLY-602), it only need to implement the default renderer(PropertySheetTable).  But, it should be aware of how this component will eventually grow into.

Reply:
> We talked about using children as a renderer and that it should default to <PropertySheetTable {...{tbl_id, highlightedRow}}/>.
>
> In the most basic case: (RowDetails of the active table): \<PropertySheet/\>

This would have been problematic because we have `detailsTblId` and `tblOptions` which are only required for table view (child) not for PropertySheet (parent which can be used with custom views as well). @robyww suggested we can create a new component combining these two for providing simple usage. Check `PropertySheetAsTable` component

> In this scenario, p is the return value of the default fetcher function where it transform the highlightedRow into an object of cname/value pairs. This object is passed in as properties to the renderer.

This is not simply name/value pairs because we also have metadata attached to columns (like unit, type, UCD, etc.) that we need to display. That's why I made parent component pass highlightedRow and tbl_id to default/table view so that it can extract information and create a table as needed. I've also added documentation for more clarity. Let me know if I missing something (I'll make changes in next PR)?

> Then, there's the more advanced use-case where fetcher is used to query external data.
In this case, fetcher is most likely async. PropertySheet should call fetcher with tbl_id, and highlightedRow so it can gather additional table meta. It should also be smart enough to render 'Loading.. ' status until fetcher returns.

Yes that's a good point. For now, I've put TODO about using fetcher, and merging this PR because it needs to go in the release this Monday. 

In next PR, I'll work on implementing a custom view using fetcher - which will make it clear for us how to evolve the existing design of these components.

Reply:
> problematic because we have detailsTblId and tblOptions

This is not a problem.  These props can be defaulted by `PropertySheetTable`, or it can be passed into from PropertySheet.  It's common for wrapper component to take the props it care about, and pass along the other props to its children without inspection.  An example of such usage is UI library wrapping input field.  e.g.  
```
{tbl_group='main', tbl_id, children, fetcher, ...passAlongProps}
```

> In this scenario, p is the return value of the default fetcher function where it transform the highlightedRow into an object of cname/value pairs. This object is passed in as properties to the renderer.

When `PropertySheet` decides to have a `renderer` and a `fetcher`, then it needs to define what that behavior should be.  I am suggesting that the behavior is.... renderer get passed in as props, `{tbl_id, highlightedRow}` and the returned value of the fetcher call.   The above comment is for when fetcher is not provided.  It would return `{tbl_id, highlightedRow}` plus the values of the highlighted row as key/value pairs.  This would satisfy the 'templating' use-case  mentioned in the ticket.  
What you are doing with `RowDetailsTable` is fine and it does not break the above contract.  It requires `tbl_id, highlightedRow` which will be given.

Because I mistakenly reviewed the PR when it was still in draft, I had to read the ticket to understand what was requested.  Based on my understanding at the time, these are my suggested design elements.  But, you are free to implement it as you see fit.  You probably have more in depth understanding of the issues and how to resolve it.

Reply:
> It's common for wrapper component to take the props it care about, and pass along the other props to its children without inspection. An example of such usage is UI library wrapping input field. e.g.

Ok this makes sense, esp the example. Although I think we have to add some sort of exception handling in this component to enforce that if children wasn't passed, then `detailsTblId` has to be passed because it's required prop for the default renderer component.

> The above comment is for when fetcher is not provided. It would return {tbl_id, highlightedRow} plus the values of the highlighted row as key/value pairs. This would satisfy the 'templating' use-case mentioned in the ticket.

Ohk I see what you mean. I gave it more thought and I think we can use a data structure that can contain metadata too besides just key-value pairs, something like this:
```
{key: {
  value: valueOfKey,
  metadata: {
    type: typeOfKey,
    units: unitsOfKey,
    ...
  },
 key2: ...
}
```

> Based on my understanding at the time, these are my suggested design elements. But, you are free to implement it as you see fit. You probably have more in depth understanding of the issues and how to resolve it.

Thanks for mentioning that. And I appreciate your input to come up with a better design.